jvm/src/test/scala/uk/gov/homeoffice/drt/protobuf/serialisation/FlightMessageConversionSpec.scala

```diff
@@ -204,40 +204,163 @@ class FlightMessageConversionSpec extends Specification {
+      actPaxOLD = Option(95),
+      tranPaxOLD = Option(10),
+      apiPaxOLD = Option(100)
+    )
+
+    val arrivalResult = FlightMessageConversion.flightMessageToApiFlight(flightMessage)
+
+    arrivalResult === apiFlight
+  }
+
+  "when flight message is deserialize and if there is actPax , transPax and apiPax present then arrival should have actPax , transPax with LiveFeedSource and apiPax with ApiFeedSource in totalPax" >> {
+    val apiFlight = arrival.copy(
+      FeedSources = Set(LiveFeedSource, AclFeedSource),
+      PassengerSources = Map(ApiFeedSource -> Passengers(Option(100), None), LiveFeedSource -> Passengers(Some(95), Some(10))),
+    )
+
+    val flightMessage = getFlightMessageWithoutPax(apiFlight).copy(
+      feedSources = Seq(LiveFeedSource.toString, AclFeedSource.toString),
+      actPaxOLD = Option(95),
+      tranPaxOLD = Option(10),
+      apiPaxOLD = Option(100)
+    )
+
+    val arrivalResult = FlightMessageConversion.flightMessageToApiFlight(flightMessage)
+
+    arrivalResult === apiFlight
+  }
+
+  "when flight message is deserialize and if there is actPax and/or transPax present then arrival should have actPax , transPax with LiveFeedSource and apiPax with ApiFeedSource in totalPax" >> {
+    val apiFlight = arrival.copy(
+      FeedSources = Set(ForecastFeedSource, AclFeedSource, LiveFeedSource),
+      PassengerSources = Map(ApiFeedSource -> Passengers(Option(100), None), LiveFeedSource -> Passengers(Option(95), Option(10))
+      ))
+
+    val flightMessage = getFlightMessageWithoutPax(apiFlight).copy(
+      feedSources = Seq(ForecastFeedSource.toString, AclFeedSource.toString, LiveFeedSource.toString),
+      actPaxOLD = apiFlight.PassengerSources.get(LiveFeedSource).flatMap(_.actual),
+      tranPaxOLD = apiFlight.PassengerSources.get(LiveFeedSource).flatMap(_.transit),
+      apiPaxOLD = Option(100)
+    )
+
+    val arrivalResult = FlightMessageConversion.flightMessageToApiFlight(flightMessage)
+
+    arrivalResult === apiFlight
+  }
+
+
+  "when flight message is deserialize and if there is actPax and/or transPax present then arrival should have actPax and transPax with AclFeedSource in totalPax" >> {
+    val apiFlight = arrival.copy(
+      FeedSources = Set(AclFeedSource),
+      PassengerSources = Map(AclFeedSource -> Passengers(Option(95), Option(10)),
+      ))
+
+    val flightMessage = getFlightMessageWithoutPax(apiFlight).copy(
+      feedSources = Seq(AclFeedSource.toString),
+      actPaxOLD = apiFlight.PassengerSources.get(AclFeedSource).flatMap(_.actual),
+      tranPaxOLD = apiFlight.PassengerSources.get(AclFeedSource).flatMap(_.transit)
+    )
+
+    val arrivalResult = FlightMessageConversion.flightMessageToApiFlight(flightMessage)
+
+    arrivalResult === apiFlight
+  }
+
+  "when arrival object has PassengerSources then serialise and deserialize should have actPax and transPax with LiveFeedSource in totalPax" >> {
```

Comment:
The descriptions of these tests say serialising and deserialising, but we're only deserialising. Would be good to do the full serialise & deserialise to check we get back what we started with